### PR TITLE
Add "Standard" object type

### PR DESCRIPTION
**Pull request by nickswalker**
## Description

Closes issue #673 

Changes proposed in this pull request:
- Specify YCB IDs for "standard objects" in this type as selected in #673
and agreed on in meeting #694
- Change references to "predefined" objects to "known" objects
to avoid confusion with "standard" objects
- Minor rephrasing and clean up

## Other comments

I have left out any description of why we chose these items or how teams can obtain them as I feel it doesn't belong in the rules ("why" is always documented here on GitHub, and they can Google information about YCB quite easily). Once we're 100% on the objects, I can take a photo of the standard objects to include in a figure.

### CODE REVIEW

File: general_rules/Scenario.tex [hunk 1]
```diff
@@ -126,46 +127,48 @@ \subsection{Changes to the arena}
...
+\subsection{Minimal Set of Known Objects}
 \label{rule:scenario_objects_list}
-The minimal configuration consists of:
+
 \begin{itemize}
 	\item \textbf{\iterm{Tableware}:} Dish, bowl, cup (or mug), and napkin.
 	\item \textbf{\iterm{Cutlery}:} Fork, knife, and spoon.
-	\item \textbf{\iterm{Trash Bags}:} Big plastic trashbags, preferrably with handle.
 	\item \textbf{\iterm{Bags}:} Lightweight. With stiff, vertical handles.
 	\item \textbf{\iterm{Disks or books}:} A set of 10 discs (LP, CD, DVD, or BluRay) or books, all of the same kind.
```
[the remark this answers is not in the document]
Comment (semaph0r) [reply general_rules/Scenario.tex]: Maybe your suggestions will delete the wrong item?

[the remark this answers is not in the document]
Comment (johaq) [reply general_rules/Scenario.tex]: Disks or books can be removed

[the remark this answers is not in the document]
Comment (semaph0r) [reply general_rules/Scenario.tex]: Darn, was a comment, no suggestion. Nevermind @johaq :smile:

File: general_rules/Scenario.tex [hunk 2]
```diff
@@ -126,46 +127,48 @@ \subsection{Changes to the arena}
...
-Assignments are announced during setup days (See~\refsec{chap:setup_and_preparation}).
-An exemplar of each object is provided before the competition for training.
 
-There are two types of objects:
+There are three types of objects:
 
 \begin{enumerate}
-	\item \textbf{\iterm{Known objects}:} Objects previously known by the robot and that it can identify and manipulate.
+	\item \textbf{\iterm{Standard objects}:} Objects announced prior to the competition.
+	These are \textit{YCB Object and Model Set} IDs 1-10, 19, 21 and 22, representing a selection of food items and household cleaners.
```
[the remark this answers is not in the document]
Comment (nickswalker) [reply general_rules/Scenario.tex]: Added

**Comment by kyordhel** [comment]
Small suggestion.

Instead of adding a new category, which are meaningful in terms of scoring and GPSR (e.g. alike objects are harder to recognize than known object), I would like to have Standard objects not as a category but as a remark. Such as:

`Objects from a standardized dataset are selected by the TC and announced X months in advance. The list of standard objects, official names, their categories, as well as information about where to acquire them can be found in \url{URL}`

This gives us the possibility of adding fruits and other types of alike objects to the list. Although we might not be able to announce a dataset for them right now (it can be generated as an @Home project) teams will know in advance which ones to start training.

**Comment by kyordhel** [comment]
From what I read, I understood that there are 3+ disjoint categories and inferred (with my team hat) that they are going to be handled differently. Needless to say how prone the rulebook can be to misinterpretations. I would like it to be as clear as possible.

This PR will do for now, but I do think we need a page in the Wiki/Website where objects are detailed. This must be done ASAP

**Comment by kyordhel** [comment]
@nickswalker please read our comments. Do change something if you see it fit, otherwise you've green light to merge.

**Comment by johaq** [comment]
I think we just need to be careful in the task remarks. See Clean Up task.

**Comment by nickswalker** [comment]
(Had to push a minor formatting fix, sorry @johaq)
These new objects don't neatly fall into the two existing categories, so I think it's necessary to create a third category for them, even though I can see how it reads strangely if there's no scoring consequence to the distinction. I think it's fine to go ahead and specify the standard objects that we know now and add additional objects in the future. 
I like the idea of linking to the year's object info (maybe we could just start the year's competition repo and push object specifics there? Something for a future PR).

**Comment by kyordhel** [comment]
@nickswalker 
Out of curiosity, why exactly do Standard Objects not fit in known/alike categories?
**Known:** Artificially manufactured objects. All instances of an object are identical.
**Alike:** Objects of the same kind that share common properties and receive the same name but are not identical or may not seem to be (like a cloth).
**Special Objects:** Anything that is used for a particular test (e.g. the bag/umbrella in carry my luggage, or the trash bin lid).

I understood all Standard objects (or most of them) would fit under known!

**Comment by nickswalker** [comment]
Hmm this is different than how I had conceived of the objects previously. I've probably been misreading the original intent of the description in the rules for years now. 

In my mind the primary difference has been about when teams get to know that an object will be used. "Known" objects would become available for use during setup (though the rules would give you a vague idea of what to expect), while "unknown" objects would be added during testing to encourage teams to handle a broad distribution of objects. Having an object known many months in advance seems distinct enough from these options to form a "super-known" category.

I think of the regular/alike distinction as an orthogonal distinction so that teams can be rewarded for handling objects that have high instance-to-instance variation (so, we could commit to having apples, then we would have some standard+alike objects)

Category    | When do teams know about them | 
-----------|----------------------------------|
Standard | (Long) before competition
Known     | Setup days, with rulebook provides minimum guidelines
Unknown | During testing              

In this framing, I think the "known" becomes slightly confusing (teams "know" about standard objects). I can't really think of an eloquent alternative. "On-site"?

(Incidentally, I also find the "alike" name confusing. "Alike" is close in meaning to "consistent" or "similar", when what we really want to highlight is difference. "Alike" objects are actually "varying").

Also curious what people's conceptions of object categories is (@RoboCupAtHome/technical-commmittee)

**Comment by johaq** [comment]
Maybe: Standard, Local, Unknown
Or: Standard, Familiar, Unfamiliar

**Comment by kyordhel** [comment]
@nickswalker You're making this unnecessarily complex. Don't overthink things.

Traditionally (since first @Home in 2006) there are two types of objects: known and unknown. Known objects are known to the robot, i.e. those the robot should be able to recognize, and the rest are unknown. If I leave my cellphone in the shelf of storing groceries, or there is an ornament there, both count as unknown. We never cared when the robot learned the object.

To make stuff slightly more complex in terms of CV, circa 2015 we came out with _alike_ objects, which are known but lack of a regular pattern. Then we had the need to add _special_ objects which are known but you are not supposed to interact with them randomly (e.g. in GPSR I won't randomly ask robots for the bag used in Help-me-Carry, or a tray that requires 2 arms). Maybe we got lost on the road, hence the confusion, so it's time to rectify.

I'd like to keep it that simple: **the robot knows an object or not**, it has a trained model that allows it to recognize the object or not, it has seen the object before or not. The rest are just sub-categories.

Why do we need unknown? Two reasons:
1. So I can place a Dr. Pepper next to a Coke and check the robot picks the Coke.
2. So I can place a pack of noodles and a bottle of beer (both unknown) and ask the robot for the square object.

In summary, we really don't need new categories, and maybe it's time to get rid of alike and special. The robot knows an object or doesn't.

**Comment by nickswalker** [comment]
I am unpersuaded. The rules don't directly control what a robot knows, they control when we release information about objects. There is expected to a relationship between how long a team knows about an object and the difficulty of interacting with it. Standard objects will behave differently than other objects in this regard, so they get a new category.

>  Standard, Familiar, Unfamiliar

I like these names better than the current ones. I'll make a PR changing these once this one is closed out

**Comment by MatthijsBurgh** [comment]
I think this one can be closed as #709 is merged.
